File: utils/Optimization/entire_path/generate_poly.m
function f = generate_poly(max_exponent,max_diff,symbol)

for k = 0:max_diff
    for i = 0:max_exponent
        if (i - k) >= 0
            f(k+1,i+1)  = factorial(i)*symbol^(i-k)/factorial(i-k);
            %f{i+1} = int(f_i,symbol,lower_limit,upper_limit);
        else 
            f(k+1,i+1) = 0;
        end
    end
end

end

File: utils/Optimization/entire_path/generate_poly.py
#! /usr/bin/env python
from math import factorial
import numpy as np
# test passed
def generate_poly(max_exponent,max_diff,symbol):
	
	f=np.zeros((max_diff+1, max_exponent+1), dtype=float)
	
	for k in range(max_diff+1):
		for i in range(max_exponent+1):
			if (i - k) >= 0:
				f[k,i]  = factorial(i)*symbol**(i-k)/factorial(i-k)
			else:
				f[k,i] = 0
	return f